support-frontend/assets/pages/contributions-landing/contributionsLanding.jsx

```diff
@@ -115,14 +122,33 @@ const router = (
         <Route
           exact
           path="/:countryId(uk|us|au|eu|int|nz|ca)/contribute/"
-          render={() => contributionsLandingPage()
-          }
+          render={(props) => {
+            const { search } = props.location;
+            const queryParams = new URLSearchParams(search);
+            const localCurrencyCountryId = queryParams.get('local-currency-country') || '';
+            const localCurrencyCountry = localCurrencyCountries[localCurrencyCountryId.toUpperCase()];
```

Comment:
Instead of doing this here and passing the `localCurrencyCountry` down through the components, we can set it in `buildInitialState` of `page.js`. This field will only be set once so I think it makes sense to initialise there.
(we already do some querystring stuff there)
That should simplify ContributionForm.jsx as well

Reply:
I've made this refactor now ✅

Reply:
nice!